addon/components/file-picker.js

```diff
@@ -73,13 +74,39 @@ export default Component.extend({
     }
   },
 
-  handleFiles: function(files) {
+  _invalidExtension: function(files) {
+    let accept = this.get('accept');
+
+    if (accept === '*') {
+      return;
+    }
+
+    let validExtensions = accept.split(',');
+
+    let fileExtensions = files.map(file => `.${file.filename.split('.').slice(-1)[0]}`);
+
+    return fileExtensions.some(extension => validExtensions.indexOf(extension) === -1);
+  },
+
+  _validate: function(files) {
     if (typeof(this.filesAreValid) === 'function') {
       if (!this.filesAreValid(files)) {
         return;
       }
     }
 
+    if (!this.get('validateExtension')) {
+      return;
+    }
+
+    return !this._invalidExtension(files);
```

Comment:
That doesn't work. If you set `validateExtensions` to `false` it will return above and return in the `handleFiles` method as well and it would be impossible to upload a file without validation.

I think what we need is:

```js
_validate: function(files) {
  if (typeof(this.filesAreValid) === 'function') {
    if (!this.filesAreValid(files)) {
      return;
    }
  }
  
  if (!this.get('validateExtensions')) {
    if (this._invalidExtension(files)) {
      return;
    }
  }
  
  return true;
}
```